### ISSUE TITLE
Missing default value in torch/optim/lr_scheduler.pyi

### ISSUE THREAD

**Issue by jeongukjae** [pytorch/pytorch#32646]
## 🐛 Bug

There is no default value in `_LRScheduler.step` in [`torch/optim/lr_scheduler.pyi#L9`](https://github.com/pytorch/pytorch/blob/5b321a098594693eca8408169f475cb334856655/torch/optim/lr_scheduler.pyi#L9), but in [`torch/optim/lr_scheduler.py#L106`](https://github.com/pytorch/pytorch/blob/5b321a098594693eca8408169f475cb334856655/torch/optim/lr_scheduler.py#L106), `step` function has default value.

---

torch/optim/lr_scheduler.pyi#L9

```python
class _LRScheduler:
    def __init__(self, optimizer: Optimizer, last_epoch: int=...) -> None: ...
    def state_dict(self) -> dict: ...
    def load_state_dict(self, state_dict: dict) -> None: ...
    def get_lr(self) -> float: ...
    def step(self, epoch: Optional[int]) -> None: ...
```

---

torch/optim/lr_scheduler.py#L106

```python
class _LRScheduler(object):
    ....
    def step(self, epoch=None):
        # Raise a warning if old pattern is detected
        # https://github.com/pytorch/pytorch/issues/20124
        if self._step_count == 1:
            if not hasattr(self.optimizer.step, "_with_counter"):
                warnings.warn("Seems like `optimizer.step()` has been overridden after learning rate scheduler "
                              "initialization. Please, make sure to call `optimizer.step()` before "
                              "`lr_scheduler.step()`. See more details at "
         ....
```

## Expected behavior

Type of `_LRScheduler` should be as follow.

```python
class _LRScheduler:
    def __init__(self, optimizer: Optimizer, last_epoch: int=...) -> None: ...
    def state_dict(self) -> dict: ...
    def load_state_dict(self, state_dict: dict) -> None: ...
    def get_lr(self) -> float: ...
    def step(self, epoch: Optional[int]=...) -> None: ...
```

cc @vincentqb

**Comment by vincentqb** [comment]
Thanks for pointing this out. Please feel free to open a PR for this.

**Comment by jeongukjae** [comment]
> Thanks for pointing this out. Please feel free to open a PR for this.

@vincentqb I created one. #32647 Would you mind if you review my PR?

**Comment by jeongukjae** [comment]
This issue is resolved by #32411
